feat(cxl-ui): cxl-featured-image add component

anoblet opened the pull request:
https://app.clickup.com/t/86b0647n0

![image](https://github.com/conversionxl/aybolit/assets/7674171/b20d68b2-4496-45ad-9f96-5e710a01d0ac)

pawelkmpt commented on packages/cxl-ui/scss/cxl-featured-image.scss:
```diff
@@ -0,0 +1,25 @@
+    bottom: 0;
+    width: 100%;
+    height: 75%;
+    background-color: var(--cxl-brand-color-black, rgb(26, 26, 26));
```
This needs using `var(--cxl-featured-image-bg-color)` which falls back to `--cxl-brand-color-black`. Different page types use different background color and it needs to be reflected here.

pawelkmpt commented on packages/cxl-ui/scss/cxl-featured-image.scss:
```diff
@@ -0,0 +1,25 @@
+:host {
+  position: relative;
+  display: block;
+  min-width: 360px;
```
```suggestion
  min-width: 320px;
```

At the same time I'm wondering whether it even needs `min-width`. It's block element which means it's 100% wide by default. And on screens smaller than 360 it causes horizontal scrollbar.

anoblet replied:
Updated. Since the element technically doesn't have any content(only a background), without setting a minimum width, it collapses to 0.

pawelkmpt commented:
One more thing, it should be visible on mobile only. Now question is whether it's better to encapsulate visibility inside the component or control it with external CSS. I don't know the answer yet

anoblet commented:
> One more thing, it should be visible on mobile only. Now question is whether it's better to encapsulate visibility inside the component or control it with external CSS. I don't know the answer yet

The correct way would be to do it from the outside, though it wouldn't be a big deal taking a shortcut here.

pawelkmpt commented on packages/cxl-ui/scss/cxl-featured-image.scss:
```diff
@@ -0,0 +1,25 @@
+    bottom: 0;
+    width: 100%;
+    height: 75%;
+    background-color: var(--cxl-featured-image-bg-color, var(--cxl-brand-color-black, rgb(26, 26, 26)));
```
We do not have `--cxl-featured-image-bg-color` defined anywhere so component background color is not compatible with what we see on the desktop.

What about refactoring this code: https://github.com/conversionxl/aybolit/blob/22f622517eb40db969d95d1300c587743fd1de65/packages/cxl-ui/scss/cxl-section.scss#L119-L147

We could set variable value instead of explicit background color value. Doable @anoblet?

pawelkmpt commented:
One more thing needed

`export { CXLFeaturedImage } from './components/cxl-featured-image';`

anoblet commented:
* Added export
* Kept `min-width` since without it, the component won't show in Storybook(Let me know if we really want to remove it)
* Set a default image source
* Added media query to hide on desktop
* Added class based background color system(https://github.com/conversionxl/aybolit/blob/fa456ce93fccfa55edebc70a08ea384bdd11572c/packages/cxl-ui/scss/cxl-featured-image.scss#L35). Let me know if this is what you intended.

pawelkmpt commented on packages/cxl-ui/scss/cxl-featured-image.scss:
```diff
@@ -0,0 +1,63 @@
+
+:host(.red) {
+  #container {
+    background-color: var(--cxl-brand-color-red);
```
```suggestion
    background-color: var(--lumo-primary-color);
```

<img width="414" alt="Screenshot 2024-06-13 at 21 40 51" src="https://github.com/conversionxl/aybolit/assets/5165721/b6231ada-090d-4b41-9737-a7a8d6eb202d">

pawelkmpt commented on packages/cxl-ui/src/components/cxl-featured-image.js:
```diff
@@ -0,0 +1,21 @@
+
+@customElement('cxl-featured-image')
+export class CXLFeaturedImage extends LitElement {
+  @property({ type: String }) src =
+    'https://ebpyu8ye7bj.exactdn.com/institute/wp-content/uploads/2023/12/ben-labay_team_headshot-1x1-bw-min-fix-min-compressed.png?strip=all&lossy=1&fit=720%2C720&ssl=1';
```
No need to set default image. Especially with such URL. Keep it empty.

pawelkmpt reviewed (changes requested):
Good approach. Just two things to fix. Please also squash all commits into one
